src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -162,69 +162,71 @@ CUDNN_SAFE_CALL(cudnnSetOpTensorDescriptor(opTensorDesc,
-                    padding_above[i] = static_cast<size_t>(padding_above_diff[i]);
-                }
 
-                if (padding_below.size() > 3)
+                if (padding_below_diff.size() > 3)
                 {
                     throw std::runtime_error(node->get_name() +
                                              "with more than 3D is not implemented.");
                 }
+
+                bool is_deconvolution = false;
                 for (auto a : data_dilation_strides)
                 {
                     if (a != 1)
                     {
-                        throw std::runtime_error(node->get_name() +
-                                                 "with data dilation is not implemented.");
+                        is_deconvolution = true;
+                        break;
                     }
                 }
 
-                bool pad_required = false;
-                if (padding_below != padding_above)
+                bool pad_required = (padding_below_diff != padding_above_diff);
+
+                Shape padding_below(padding_below_diff.size(), 0);
+                Shape padding_above(padding_above_diff.size(), 0);
+                for (int i = 0; i < padding_below.size(); i++)
                 {
-                    pad_required = true;
+                    padding_below[i] = static_cast<size_t>(padding_below_diff[i]);
+                    padding_above[i] = static_cast<size_t>(padding_above_diff[i]);
                 }
-                auto input_shape = args[0].get_shape();
-                auto input_shape_padded = input_shape;
 
-                if (pad_required)
+                auto input_shape = args[0].get_shape();
+                Shape input_shape_padded = input_shape;
+                Shape padding_interior(data_dilation_strides);
+                writer.block_begin("  // " + node->get_name());
+                if (pad_required || is_deconvolution)
                 {
-                    input_shape_padded =
-                        get_padded_shape(input_shape, padding_below, padding_above, {});
+                    input_shape_padded = get_padded_shape(
+                        input_shape, padding_below, padding_above, padding_interior);
+                    Shape input_padded_strides = row_major_strides(input_shape_padded);
                     auto temp_size =
                         shape_size(input_shape_padded) * args[0].get_element_type().size();
                     GPUAllocator allocator =
                         external_function->get_primitive_emitter()->get_memory_allocator();
                     size_t idx_workspace = allocator.reserve_workspace(temp_size);
                     writer << "void* pad_buffer = runtime::gpu::invoke_memory_primitive(ctx, "
                            << idx_workspace << ");\n";
+                    writer << "std::vector<" << args[0].get_type() << "> pad_buffer_host("
+                           << shape_size(input_shape_padded) << ", 0);\n";
```

Comment:
Will the pad value always be zero? `build_pad` supports padding value other than 0.

Reply:
it can be set to any value.

Reply:
Should it be part of the kernel?

Reply:
I can be. as described above, there are two methods to do it. I not sure which one is better yet.